torch/csrc/jit/codegen/cuda/lower_sync_information.cpp

```diff
@@ -213,7 +213,13 @@ void SyncMap::build(Fusion* fusion) {
           // If consumer is parallelized with this type but producer is
           //  predicated redundant on this type. This parallel dimension
           //  is a RAW dimension. See test: FusionSeriaSmemWriteParallelRead1/2
-          if (c_id != nullptr && producer_redundant_types.get(parallel_type)) {
+          //
```

Comment:
Is this specific to this PR?

Reply:
Can you also please add a test for this specific pattern?

Reply:
Does this mean whenever we have a redundant op, we would insert a sync immediately after that? Then, what would happen if we have a chain of redundant exprs? In my naive quick thinking, it seems we should just add a sync when an expr chain becomes non redundant.

Reply:
Why don't we extract this change out of this PR?

Reply:
The sync insertion wouldn't be right after the redundant op, which I believe currently only means redundant write. The usual RAW sync insertion rule still applies here, i.e. a sync right before they are used.

As on the comment and TODO, this sync wouldn't be needed if all uses of this redundant write ends with another redundant shared/global write with the same redundant type. This case would need an additional traversal info somewhere. 

As this issue currently lives in devel, just wanted to do a quick patch and then follow up.

Reply:
Removed from this PR and moved to #1684